### PR TITLE
[Torch] Support mask mod arguments in flex_attention HOP

### PR DESCRIPTION
**Pull request by keshavvinayak01**
PyTorch flex_attention passes mask_mod closure captures as mask_mod_other_buffers, but torch.hop_flex_attention only modeled the mask function symbol. This adds trailing variadic mask_mod_arguments operands so explicit attention masks and other captures can be represented in Torch IR.

The FX importer forwards mask_mod_other_buffers into those operands while keeping score_mod_fn and mask_mod_fn as symbol attributes. Tests cover textual round-tripping and importer emission for an explicit attention mask capture.

Tests: cmake --build /home/keshavvinayakjha/Desktop/code/iree-worktrees/torch-mlir-flex-mask-mod-args/build --target torch-mlir-opt; llvm-lit -sv /home/keshavvinayakjha/Desktop/code/iree-worktrees/torch-mlir-flex-mask-mod-args/build/test/Dialect/Torch/ops.mlir; PyTorch export smoke test for HOP mask_mod_other_buffers.

### CODE REVIEW

**Comment by keshavvinayak01** [comment]
AI agent messed up haha.
